bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -0,0 +1,221 @@
+                                if (liveCopy != null) {
+                                    root = liveCopy.getBlueprintPath();
+                                }
+                            }
+                        } else if (relationshipManager.hasLiveRelationship(resource)) {
+                            // the resource is a live copy
+                            LiveRelationship liveRelationship = relationshipManager.getLiveRelationship(resource, false);
+                            if (liveRelationship != null) {
+                                LiveCopy liveCopy = liveRelationship.getLiveCopy();
+                                if (liveCopy != null) {
+                                    root = liveCopy.getPath();
+                                }
+                            }
+                        }
+                    } catch (WCMException e) {
+                        // ignore
+                    }
+                }
+            }
+        }
+        return root;
+    }
+
+    /**
+     * Returns the localization root of the experience fragment path based on the localization root of the current page.
+     *
+     * As of today (08/aug/2019) the XF UI does not support creating Live and Language Copies, which prevents getRoot
+     * to be used with XF.
+     * This method works around this issue by deducting the XF root from the XF path and the root of the current page.
+     *
+     * @param xfPath The experience fragment path.
+     * @param currentPageRoot The localization root of the current page.
+     * @return The localization root of the experience fragment path if it exists, {@code null} otherwise.
+     */
+    private String getXfRoot(String xfPath, String currentPageRoot) {
+        String xfRoot = null;
+        if (!StringUtils.isEmpty(xfPath) && !StringUtils.isEmpty(currentPageRoot)
+            && resolver.getResource(xfPath) != null && resolver.getResource(currentPageRoot) != null) {
+            String[] xfPathTokens = Text.explode(xfPath, PATH_DELIMITER_CHAR);
+            String[] referenceRootTokens = Text.explode(currentPageRoot, PATH_DELIMITER_CHAR);
+            int xfRootDepth = referenceRootTokens.length + 1;
+            if (xfPathTokens.length >= xfRootDepth) {
+                String[] xfRootTokens = new String[xfRootDepth];
+                System.arraycopy(xfPathTokens, 0, xfRootTokens, 0, xfRootDepth);
+                xfRoot = StringUtils.join(PATH_DELIMITER, Text.implode(xfRootTokens, PATH_DELIMITER));
+            }
+        }
+        return xfRoot;
+    }
+
+    /**
+     * Checks if the resource exists at the given path.
+     *
+     * @param path The resource path.
+     * @return {@code true} if the resource exists, {@code false} otherwise.
+     */
+    private boolean resourceExists(String path) {
+        if (StringUtils.isEmpty(path)) {
+            return false;
+        }
+        return (resolver.getResource(path) != null);
```

Comment:
could be simplified to `StringUtils.isNotEmpty(path) && resolver.getResource(path) != null`

Reply:
Good catch! This was addressed in https://github.com/adobe/aem-core-wcm-components/pull/675/commits/7bce91c9a620b6d169695210a9d0e07347369a2e